distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/config/UrlResourceReaderConfigurator.java

```diff
@@ -35,13 +36,29 @@ public UrlResourceReaderConfigurator(ConfigurationAdmin configAdmin) {
     this.configAdmin = configAdmin;
   }
 
+  @SuppressWarnings("squid:S2142")
   public void setUrlResourceReaderRootDirs(String... rootResourceDirs) throws IOException {
     Configuration configuration = configAdmin.getConfiguration(PID, null);
     Dictionary<String, Object> properties = new DictionaryMap<>();
     Set<String> rootResourceDirectories =
         ImmutableSet.<String>builder().add(rootResourceDirs).build();
     properties.put("rootResourceDirectories", rootResourceDirectories);
     configuration.update(properties);
+    for (int i = 0; i < 5; i++) {
+      Configuration updatedConfig = configAdmin.getConfiguration(PID, null);
+      if (updatedConfig
+          .getProperties()
+          .get("rootResourceDirectories")
+          .equals(rootResourceDirectories)) {
+        break;
+      } else {
+        try {
+          TimeUnit.SECONDS.sleep(5);
+        } catch (InterruptedException e) {
+          // ignore
```

Comment:
❗️ Interruptions should not be simply ignored. We need to remember that an interruption occurred from within and propagate it as part of a try/finally outside of the loop.

For example: ```try {
  for (...) {
    try {
      TimeUnits.SECONDS.sleep(5);
    } catch (InterruptedException e) {
     interrupted = true;
    }
  }
} finally { 
  if (interrupted) {
    Thread.currentThread().interrupt();
  }
}```

Reply:
I was ignoring it because this is test code and I don't really care about them.